ChallengeBugTracking/Challenge3.cs

```diff
@@ -21,15 +22,17 @@ public class Leaf : Node {
     public static string[] FlattenArray(IEnumerable<Node> nodes) {
         // TODO: ISSUE #4242 : Our customer reported that this method doesn't work in the case of deeply nested arrays
         // I added the test case to the test file, but I'm not sure how to fix it. Can you help?
-        var result = new List<string>();
+        List<string> Output = new List<String>();
         foreach(var node in nodes) {
             if(node is Branch b) {
-                result.AddRange(b.Children.Cast<Leaf>().Select(l => l.Value));
+                Output.AddRange(b.Children.Where(o => o is Leaf).Cast<Leaf>().Select(l => l.Value));
+                if (b.Children.Any(ot => ot is Branch)) {
+                    Output.AddRange(FlattenArray(b.Children.Where(r => r is Branch)));
+                }
             } else if(node is Leaf l) {
-                result.Add(l.Value);
+                Output.Add(l.Value);
             }
         }
-
-        return result.ToArray();
+        return Output.ToArray();
```

Comment:
Your code returns `["this","is","a","nested","super","array"]`
Thank you for helping me fixing my tests, I will push a fix shortly. Can you rebase your branch on it ?